src/PrestaShopBundle/Resources/config/services/bundle/repository.yml

```diff
@@ -138,8 +138,12 @@ services:
     alias: PrestaShopBundle\Entity\Repository\AttributeGroupRepository
     deprecated: ~
 
-  prestashop.core.admin.feature_flag.repository:
-    class: PrestaShopBundle\Entity\Repository\FeatureFlagRepository
+  PrestaShopBundle\Entity\Repository\FeatureFlagRepository:
     factory: [ '@doctrine.orm.default_entity_manager', getRepository ]
     arguments:
       - PrestaShopBundle\Entity\FeatureFlag
+    public: true
```

Comment:
why public?

Reply:
oh cuz its used in controller.
For future when we need services that are private in controller, how would we do it? Maybe its time to do it in this case? (I think when controllers are not autowirded such services can be added into subscribed services list and used that way)

Reply:
Yep, the service should be injected in the action when it's specific to only one action, in the constructor (but @FabienPapet mentioned it as not a good practice) or ideally vie the subscribed services feature indeed. But I believe some stuffs prevent us to do this cleanly, this should be fixed in v9 and then we can start correctly handling these injected services

Reply:
Once all this is clear we'll be able to improve our services so that most of them are private